### PR TITLE
Add multiplayer spectating user state and button

### PR DESCRIPTION
**Pull request by smoogipoo**
This adds the `MultiplayerUserState.Spectating` state, the spectate button, the logic which toggles the spectating state and the ready button, and the footer design change to accommodate the spectate button.

The `Spectating` user state is a constant state - once you have it, the multiplayer server will not switch you out from that state by itself. The user can go back into the idle state themselves when they want.

Although I'm still finalising the osu-server-spectator changes, the `Spectating` state is pretty much just used for signalling other clients, with the rest of the logic such as entering gameplay being handled client-side. The only exception to this is a special case for a spectating host which is allowed to start the match even though they aren't in the `Ready` state.

To keep things simple I've currently made the operation a no-op in the SignalR `MultiplayerClient`, so that errors aren't spewed to the client but tests keep working.

Can be merged as is, or waiting for the other 800+ LOC actually implementing the spectator screen.

### CODE REVIEW

**Comment by peppy** [comment]
@smoogipoo gonna grab the test failures for this? they look like probable regressions.

**Comment by smoogipoo** [comment]
Yes I'll get them

File: osu.Game/Online/Multiplayer/MultiplayerClient.cs [hunk 1]
```diff
@@ -96,6 +96,9 @@ public override Task ChangeState(MultiplayerUserState newState)
             if (!IsConnected.Value)
                 return Task.CompletedTask;
 
+            if (newState == MultiplayerUserState.Spectating)
+                return Task.CompletedTask; // Not supported yet.
```
Comment (peppy) [inline osu.Game/Online/Multiplayer/MultiplayerClient.cs:100]: I guess this is here to avoid an exception returning from the server?

Comment (smoogipoo) [reply osu.Game/Online/Multiplayer/MultiplayerClient.cs:100]: Correct

File: osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerReadyButton.cs [hunk 2]
```diff
@@ -105,6 +107,13 @@ private void updateState()
 
             button.Enabled.Value = Client.Room?.State == MultiplayerRoomState.Open && !operationInProgress.Value;
 
+            // When the local user is the host and spectating the match, the "start match" state should be enabled.
+            if (localUser.State == MultiplayerUserState.Spectating)
+            {
+                button.Enabled.Value &= Room?.Host?.Equals(localUser) == true;
+                button.Enabled.Value &= newCountReady > 0;
+            }
+
```
Comment (peppy) [inline osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerReadyButton.cs:116]: Not sure about doing this in multiple steps... means the button could enter enabled and then disabled in the same call to this method. Probably best extracting a bool out and setting it to `button.Enabled.Value` at the end?

**Review by peppy — CHANGES_REQUESTED** [review]
Looks pretty solid apart from the one issue 👍🏻

**Review by peppy — APPROVED** [review]

**Pull request merged**